### PR TITLE
Parameter file updates: fire, logging, hydro, seeds and fates_scalar

### PR DESCRIPTION
**Pull request by rgknox**
### Description:

This changeset incorporates a group of changes to the fates parameter file. This includes:

1) Updating the naming convention for fire parameters
2) Adding the export fraction parameter for logging
3) Adding the absorbing root surface conductivity for root-to-soil flow parameter (kmax_rsurf2)
4) Changing seed_rain to seed_suppl
5) Remove pft_used
6) Remove fates_scalar dimension

**Note: This changes file compatibility, not just values. Users must update their personal files to this format to be compatible with tags moving forward.**

**Note: This is an API changing pull request, requiring updated default files for the HLMs**

### Collaborators:

@jkshuman @huangmy @xuchongang @lmkueppers @ekluzek 

### Checklist:

- [] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run

### Test Results:

TBD

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

**Comment by rgknox** [comment]
this replaces #513

**Comment by rgknox** [comment]
updating logging_code to a double precision, see https://github.com/ESCOMP/ctsm/issues/701

**Comment by jkshuman** [comment]
@rgknox the parameter names are updated with the prefix, but the refactor (re-ordering) of the values has been lost for a number of the fire parameters. The correct values are in the fates_params_14pfts.cdl in my SF_fuels_refactor branch
https://github.com/jkshuman/fates/blob/SF_fuels_refactor/parameter_files/fates_params_14pfts.cdl

Are you able to pull these, or where should I update them?

**Comment by jkshuman** [comment]
Updating these values now in the PR. For some reason I didn't see the diff of the cdl file.

File: parameter_files/fates_params_default.cdl [hunk 1]
```diff
@@ -523,147 +351,251 @@ variables:
...
+		fates_cwd_flig:long_name = "Lignin fraction of coarse woody debris" ;
+	float fates_fire_drying_ratio ;
+		fates_fire_drying_ratio:units = "NA" ;
+		fates_fire_drying_ratio:long_name = "spitfire parameter, fire drying ratio for fuel moisture, alpha_FMC EQ 6 Thonicke et al 2010" ;
+	float fates_fire_durat_slope ;
+		fates_fire_durat_slope:units = "NA" ;
+		fates_fire_durat_slope:long_name = "spitfire parameter, fire max duration slope, Equation 14 Thonicke et al 2010" ;
+	float fates_fire_fdi_a ;
+		fates_fire_fdi_a:units = "NA" ;
+		fates_fire_fdi_a:long_name = "spitfire parameter (unknown) " ;
```
Comment (jkshuman) [inline parameter_files/fates_params_default.cdl]: fates_fire_fdi_a:long_name = "spitfire parameter, fire danger index  Equation 5 Thonicke et al 2010" ;

**Comment by rgknox** [comment]
@jkshuman, the order is the result of the sorting algorithm. It first orders by dimension, and then orders by alphabetical order.  Perhaps we should open up an issue on how the sorting script works?

**Comment by jkshuman** [comment]
@rgknox it is the actual numbers that needed to be reordered for many of the parameters. The grasses are both at the end, so the associated values needed to be moved around. I have a review going with the corrections. Those updated values in the parameter file must have been lost in the merge. They were in my branch for the SF_fuels_refactor, so I am just updating them from that in the PR.

File: parameter_files/fates_params_default.cdl [hunk 2]
```diff
@@ -523,147 +351,251 @@ variables:
...
+		fates_fire_max_durat:units = "minutes" ;
+		fates_fire_max_durat:long_name = "spitfire parameter, fire maximum duration, Equation 14 Thonicke et al 2010" ;
+	float fates_fire_miner_damp ;
+		fates_fire_miner_damp:units = "NA" ;
+		fates_fire_miner_damp:long_name = "spitfire parameter, mineral-dampening coefficient EQ A1 Thonicke et al 2010 " ;
+	float fates_fire_miner_total ;
+		fates_fire_miner_total:units = "fraction" ;
+		fates_fire_miner_total:long_name = "spitfire parameter, total mineral content, Table A1 Thonicke et al 2010" ;
+	float fates_fire_nignitions ;
+		fates_fire_nignitions:units = "/m2 (?)" ;
```
[the remark this answers is not in the document]
Comment (ckoven) [reply parameter_files/fates_params_default.cdl]: what's the areal denominator of the units for `fates_fire_nignitions`? per gridcell or per some specific area?

[the remark this answers is not in the document]
Comment (jkshuman) [reply parameter_files/fates_params_default.cdl]: @ckoven  In the calculation currentPatch%NF it is then scaled by the area, so it ends up being ignitions per day per km2.

[the remark this answers is not in the document]
Comment (jkshuman) [reply parameter_files/fates_params_default.cdl]: @ckoven probably should add that back in, so as people create ignition values they are at same spatial scale...is that your thought?

**Comment by ckoven** [comment]
I think we may need to modify both the ncvarsort.py and the modify_fates_paramfile.py tools as part of this PR, now that we are losing the `fates_scalar` dimension.
e.g., see specific references to it in these places:
https://github.com/NGEET/fates/blob/master/tools/modify_fates_paramfile.py#L89
https://github.com/NGEET/fates/blob/master/tools/ncvarsort.py#L38

File: parameter_files/fates_params_default.cdl [hunk 3]
```diff
@@ -1111,53 +988,191 @@ data:
   _, _, _, _, _, _, _, _, _, _, _, _,
   _, _, _, _, _, _, _, _, _, _, _, _ ;
 
- fates_turnover_retrans_mode = 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 ;
+ fates_history_height_bin_edges = 0, 0.1, 0.3, 1, 3, 10 ;
 
- fates_wood_density = 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 0.7, 
-    0.7 ;
+ fates_fire_FBD = 4, 15.4, 16.8, 19.6, 999, 4 ;
```
[the remark this answers is not in the document]
Comment (rgknox) [reply parameter_files/fates_params_default.cdl]: @jkshuman , aren't these changes being implemented in PR #523 ?

[the remark this answers is not in the document]
Comment (jkshuman) [reply parameter_files/fates_params_default.cdl]: @rgknox you are 100% correct this is for PR #523 that would explain why they were left out...
are you able to just ignore my value changes?

[the remark this answers is not in the document]
Comment (rgknox) [reply parameter_files/fates_params_default.cdl]: I think so.

**Comment by rgknox** [comment]
See: 
https://github.com/NGEET/fates/issues/529#issuecomment-489706271

I think I would like to add fates_q10_mr and fates_froz_q10 in this iteration as well.

**Comment by ckoven** [comment]
I was about to issue a PR to this branch, updating the parameter file tools to work properl;y on this now that we are getting gotten rid of the fates_scalar dimension, but in testing those changes I'm finding that ncvarsort.py doesn't work at all, in either the base or my modified version.  Could someone else test that to see what  the deal is?  maybe the scipy netcdf doesn't work the same way as the netcdf4 library and it hasn't been working properly ever since that change?

**Comment by ckoven** [comment]
I support pulling out the q10 stuff from that synchronization machinery.  there is absolutely no reason why those need to be synchronized with the host.  also happy to get rid of the frozen_q10 logic entirely for the CWD breakdown.

**Comment by rgknox** [comment]
weird, I used the sort tool to generate the current file in this PR.

**Comment by ckoven** [comment]
does it actually sort the variables when you use it?

**Comment by rgknox** [comment]
I think so, I just looked through the file and it looks sorted correctly to me.  Can you verify?  Note in particular that the scalar variables seem sorted as well.

**Comment by ckoven** [comment]
maybe i did it wrong?  i'll check again.

**Comment by jkshuman** [comment]
@rgknox let me know if you want me to look this over again at some point. Thanks for catching my brain freeze/merge of those two PRs. Do you want me to remove those value suggestions, or just leave it be at this point?

**Comment by rgknox** [comment]
Either is fine, thanks @jkshuman

**Comment by rgknox** [comment]
Updated fates_max_decomp per #533

**Comment by rgknox** [comment]
I would like to remove the following parameter, which is not used:

https://github.com/NGEET/fates/blob/master/parameter_files/fates_params_default.cdl#L457

Also, I will bring over the q10 parameters mentioned here: https://github.com/NGEET/fates/issues/529

**Comment by rgknox** [comment]
@ckoven , I just ran a test.  I moved fates_hydr_p50_node and fates_recruit_hgt_min out of order.  Running the ncvarsort moved them back to the correct placement in the new file structure.

**Comment by huangmy** [comment]
@rgknox I reviewed the changes related to the logging export fraction, in general they look great to me. Just want to make sure that the hard-coded temporary export fraction is now removed, and you have added this parameter in to the python script. If these are done, I have no other comments.

**Comment by rgknox** [comment]
@huangmy , right, thanks  for the reminder! That will now be switched over.

**Comment by rgknox** [comment]
Conducted regression tests with a version that temporarily used 4 byte reals in the parameter file (pre-existing precision).  Results match baseline.

/gpfs/fs1/scratch/rgknox/clmed-tests/fates-api-7.3.0-floats-C6767148-Fd259955.fates.cheyenne.intel

Will re-test with double precision (no baseline comparison).

**Comment by rgknox** [comment]
tests with double precision parameter files also pass:
/gpfs/fs1/scratch/rgknox/clmed-tests/fates-sci.1.27.1_api.7.3.0-C6767148-Fd259955.fates.cheyenne.intel

**Pull request merged**